### ISSUE TITLE
Unsaved changes screen fails to load on quit from menu

### ISSUE THREAD

**Issue by MineRobber9000** [nesbox/TIC-80#1969]
**How to reproduce:**
1. Start TIC-80 1.0.
2. Make a change to the default cart.
3. Run the cart.
4. Enter the menu with the cart running.
5. Select `QUIT TIC-80`

**Expected behavior:**
The unsaved changes screen appears, asking me if I really want to exit with unsaved changes.

**What actually happens:**
The menu seemingly disappears.

I say "seemingly" because pressing either A or B reveals the previous menu to still be active, albeit hidden from view. Pressing B at this stage will resume the game.

**Comment by sogaiu** [comment]
I reproduced with 2a662a0ad38514210c313b518f7691aab227c06c.

Here's what I did:

1. Start tic80
2. Press esc to get to the editor
3. Make a change (e.g. t=1 instead of t=0)
4. Press ctrl-r
5. Press esc
6. Choose "QUIT TIC-80"
7. Observe menu disappear but notice still in "the zone"

Pressing esc at this point gets me back to the game (though very briefly I see the menu appear before I see the game again).

If I return to the menu again and choose "CLOSE GAME" and then enter "quit" followed by the enter key, I see the message about unsaved changes.  I can exit by choosing "YES" with no problem.
